### PR TITLE
[move-prover] functional tests for strong edges

### PR DESCRIPTION
**Pull request by JakeSilverman**
## Motivation

Add functional tests for strong edges

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

File: language/move-prover/tests/sources/functional/strong_edges.move [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        aborts_if len(v) == 0;
+        ensures v[0] == 7;
+        ensures v[1] == old(v[1]);
+    }
+
+    struct R has copy, drop {
+        x: u64
+    }
+
+    public fun splice_edge_helper(r_ref: &mut R) : &mut u64 {
```
Comment (shazqadeer) [inline language/move-prover/tests/sources/functional/strong_edges.move:52]: What specific aspect of strong edges are you testing with this example?

Comment (JakeSilverman) [reply language/move-prover/tests/sources/functional/strong_edges.move:52]: you're right... this is testing a weak edge, albeit one that can be seen in the strong edge analysis. I was looking for coverage of the strong edge analysis (as opposed to just of strong edges). The name of the file (strong_edges) is misleading. Let me know if you want me to remove this test case or change name of file.

**Comment by JakeSilverman** [comment]
I removed the flag from top of the test file to avoid having to submit new PR after https://github.com/diem/diem/pull/8024 lands.

**Review by wrwg — APPROVED** [review]

**Pull request merged**